components/tools/OmeroJava/test/integration/LightAdminRolesTest.java

```diff
@@ -899,6 +984,91 @@ public void testLinkNoSudo(boolean permWriteOwned, boolean permChown,
+     * @throws Exception unexpected
+     */
+    @Test(dataProvider = "WriteOwned and IsAdmin cases")
+    public void testLinkMemberOfGroupNoSudo(boolean permWriteOwned, boolean isAdmin,
+            String groupPermissions) throws Exception {
+        /* WriteOwned permission is necessary and sufficient for lightAdmin to link
+         * others objects to their objects. Exceptions are Private group, where such linking will
+         * fail in all cases and Read-Write group where linking will succeed even
+         * for otherUser (otherUser and lightAdmin are both members of the group).*/
+        boolean isExpectLinkingSuccessAdmin = (permWriteOwned && !groupPermissions.equals("rw----") || groupPermissions.equals("rwrw--"));
+        boolean isExpectLinkingSuccessUser = groupPermissions.equals("rwrw--");
+        final EventContext normalUser = newUserAndGroup(groupPermissions);
+        final EventContext otherUser = newUserAndGroup(groupPermissions);
+        ExperimenterGroup normalUsergroup = new ExperimenterGroupI(normalUser.groupId, false);
+        /* Set up the light admin's permissions for this test.*/
+        List<String> permissions = new ArrayList<String>();
+        if (permWriteOwned) permissions.add(AdminPrivilegeWriteOwned.value);
+        final EventContext lightAdmin = loginNewAdmin(true, permissions);
+        /* root adds lightAdmin to normalUser's group.*/
+        logRootIntoGroup(normalUser);
+        normalUsergroup = addUsers(normalUsergroup, ImmutableList.of(lightAdmin.userId, otherUser.userId), false);
+        /* Create Dataset and Project as normalUser in normalUser's group.*/
+        loginUser(normalUser);
+        client.getImplicitContext().put("omero.group", Long.toString(normalUser.groupId));
+        Dataset dat = mmFactory.simpleDataset();
+        Dataset sentDat = (Dataset) iUpdate.saveAndReturnObject(dat);
+        Project proj = mmFactory.simpleProject();
+        Project sentProj = (Project) iUpdate.saveAndReturnObject(proj);
+        /* Create Image and Dataset as lightAdmin or otherUser in normalUser's group.*/
+        if(isAdmin) {
+            loginUser(lightAdmin);
+        } else {
+            loginUser(otherUser);
+        }
+        client.getImplicitContext().put("omero.group", Long.toString(normalUser.groupId));
+        Image ownImage = mmFactory.createImage();
+        Image sentOwnImage = (Image) iUpdate.saveAndReturnObject(ownImage);
+        Dataset ownDat = mmFactory.simpleDataset();
+        Dataset sentOwnDat = (Dataset) iUpdate.saveAndReturnObject(ownDat);
+        /* lightAdmin or otherUser checks that the canLink value on all the objects to be linked
+         * is true (for own image) and for other people's objects (sentProj, sentDat) the canLink
+         * are matching the expected behavior (see booleans isExpectLinkingSuccess... definitions).*/
+        Assert.assertTrue(getCurrentPermissions(sentOwnImage).canLink());
+        if (isAdmin) {
```

Comment:
I wonder if you can do something like,

```java
final boolean isExpectLinkingSuccess = isAdmin ? isExpectLinkingSuccessAdmin : isExpectLinkingSuccessUser;
```

then use it to simplify the following if/else's.